tsdb/head_append.go

```diff
@@ -1012,185 +1028,178 @@ func (a *headAppender) Commit() (err error) {
+	// WBL Graphically:
+	//	WBL Before this Commit(): [old samples before this commit for chunk 1]
+	//	WBL After this Commit():  [old samples before this commit for chunk 1][new samples in this commit for chunk 1]mmapmarker1[samples for chunk 2]mmapmarker2[samples for chunk 3]
+	if acc.oooMmapMarkers != nil {
+		markers := make([]record.RefMmapMarker, 0, acc.oooMmapMarkersCount)
+		for ref, mmapRefs := range acc.oooMmapMarkers {
+			for _, mmapRef := range mmapRefs {
+				markers = append(markers, record.RefMmapMarker{
+					Ref:     ref,
+					MmapRef: mmapRef,
+				})
 			}
-			r := enc.MmapMarkers(markers, a.head.getBytesBuffer())
-			oooRecords = append(oooRecords, r)
 		}
+		r := acc.enc.MmapMarkers(markers, a.head.getBytesBuffer())
+		acc.oooRecords = append(acc.oooRecords, r)
+	}
 
-		if len(wblSamples) > 0 {
-			r := enc.Samples(wblSamples, a.head.getBytesBuffer())
-			oooRecords = append(oooRecords, r)
-		}
-		if len(wblHistograms) > 0 {
-			r := enc.HistogramSamples(wblHistograms, a.head.getBytesBuffer())
-			oooRecords = append(oooRecords, r)
-		}
-		if len(wblFloatHistograms) > 0 {
-			r := enc.FloatHistogramSamples(wblFloatHistograms, a.head.getBytesBuffer())
-			oooRecords = append(oooRecords, r)
-		}
+	if len(acc.wblSamples) > 0 {
+		r := acc.enc.Samples(acc.wblSamples, a.head.getBytesBuffer())
+		acc.oooRecords = append(acc.oooRecords, r)
+	}
+	if len(acc.wblHistograms) > 0 {
+		r := acc.enc.HistogramSamples(acc.wblHistograms, a.head.getBytesBuffer())
+		acc.oooRecords = append(acc.oooRecords, r)
+	}
+	if len(acc.wblFloatHistograms) > 0 {
+		r := acc.enc.FloatHistogramSamples(acc.wblFloatHistograms, a.head.getBytesBuffer())
+		acc.oooRecords = append(acc.oooRecords, r)
+	}
+
+	acc.wblSamples = nil
+	acc.wblHistograms = nil
+	acc.wblFloatHistograms = nil
+	acc.oooMmapMarkers = nil
+}
 
-		wblSamples = nil
-		wblHistograms = nil
-		wblFloatHistograms = nil
-		oooMmapMarkers = nil
+// handleAppendableError processes errors encountered during sample appending and updates
+// the provided counters accordingly.
+//
+// Parameters:
+// - err: The error encountered during appending.
+// - appended: Pointer to the counter tracking the number of successfully appended samples.
+// - oooRejected: Pointer to the counter tracking the number of out-of-order samples rejected.
+// - oobRejected: Pointer to the counter tracking the number of out-of-bounds samples rejected.
+// - tooOldRejected: Pointer to the counter tracking the number of too-old samples rejected.
+//
+// Behavior:
+// - If err is nil, no action is taken.
+// - If err is storage.ErrOutOfOrderSample, the appended counter is decremented and the oooRejected counter is incremented.
+// - If err is storage.ErrOutOfBounds, the appended counter is decremented and the oobRejected counter is incremented.
+// - If err is storage.ErrTooOldSample, the appended counter is decremented and the tooOldRejected counter is incremented.
+// - For any other error, the appended counter is decremented.
```

Comment:
Nit: I think we can skip this `Behavior` bit in the docs because the code is self explanatory.